### ISSUE TITLE
`uvflag` string handling update makes it break backwards compat

### ISSUE THREAD

**Issue by nkern** [RadioAstronomySoftwareGroup/pyuvdata#667]
Apparently old `UVFlag` objects (older than Aug 2019) wrote their polarization arrays as a string, (e.g. `array([b'-5,-6'], dtype='|S5')`), which was parsed as `split(',')` upon read-in. An update to UVFlag makes this an integer array as it should have been, but didn't provide backwards compatability. Therefore all of the `IDR2` UVFlag objects are no longer readable. Probably backwards compat should be supported for a little while for such a large API change

**Comment by mkolopanis** [comment]
These polarization arrays exist as strings like this if `collapse_pol` has been called, and existed before too. Interesting that it is having an issue reading an older save file, we use one of these in testing too. perhaps it has not had pols collapsed.

**Comment by nkern** [comment]
sorry I updated the main conversation with the traceback to be more specific. oddly, it works for a single file, but not when it goes to read multiple files

**Comment by mkolopanis** [comment]
this is interesting, but I think probably is a small overlook in the `expected_type` of the polarization attribute.

**Comment by mkolopanis** [comment]
thanks for bringing it up.

**Comment by nkern** [comment]
Okay so here are some more hints:
If I do 
`uvf = UVFlag(ffiles[:1])` I get no error, but it then fails its check
`uvf.check()` yields the same dtype error. Also `uvf.polarization_array` yields `array(['-5,-6'], dtype='<U5')`, which seems wrong. I think the issue with multiple files is that before concatenating the `check` is called. Seems like `uvf` check should be called upon read-in (for a single file) with that kwarg exposed to the init function (kind of like `run_check` is exposed in `UVData.read`)

**Comment by mkolopanis** [comment]
The check error I think is expected. Well I expect the error but it _shouldn't_ error. This is a quick fix. the expected type is not probably set when reading files which have had their polarization collapsed

**Comment by nkern** [comment]
But I haven't collapsed their polarizations (I just tried to read the file, that's it). I want the `polarization_array` to be integers, not a string.

Also seems like in the case of a single filepath, _it should fail_ but it doesn't b/c `self.check()` isn't called at the end of the read, and I think by default it should, but we should have the option to turn it off (just like UVData.read)

**Comment by mkolopanis** [comment]
Can you paste in the history from the single file with the string polarization array please?

**Comment by nkern** [comment]
Yup, its `Flag object with type "antenna" created by 1.4.0.  Git origin: https://github.com/HERA-Team/pyuvdata.git.  Git hash: 7543347ea10852f249e63d17297712781a1cdcdb.  Git branch: master.  Git description: v1.4-85-g7543347e.Collapsed to type "waterfall" with 1.4.0.  Git origin: https://github.com/HERA-Team/pyuvdata.git.  Git hash: 7543347ea10852f249e63d17297712781a1cdcdb.  Git branch: master.  Git description: v1.4-85-g7543347e.Pol axis collapsed with 1.4.0.  Git origin: https://github.com/HERA-Team/pyuvdata.git.  Git hash: 7543347ea10852f249e63d17297712781a1cdcdb.  Git branch: master.  Git description: v1.4-85-g7543347e.Collapsed to type "waterfall" with 1.4.0.  Git origin: https://github.com/HERA-Team/pyuvdata.git.  Git hash: 7543347ea10852f249e63d17297712781a1cdcdb.  Git branch: master.  Git description: v1.4-85-g7543347e.Converted to mode "flag" with 1.4.0.  Git origin: https://github.com/HERA-Team/pyuvdata.git.  Git hash: 7543347ea10852f249e63d17297712781a1cdcdb.  Git branch: master.  Git description: v1.4-85-g7543347e.Written by 1.4.0.  Git origin: https://github.com/HERA-Team/pyuvdata.git.  Git hash: 7543347ea10852f249e63d17297712781a1cdcdb.  Git branch: master.  Git description: v1.4-85-g7543347e.  Read/written with pyuvdata version: 1.4.1.  Git origin: https://github.com/HERA-Team/pyuvdata.  Git hash: f208c9a1cbee1afdebfc8f5b41e2c119b34d37a6.  Git branch: master.  Git description: v1.3-137-gf208c9a.`

**Comment by mkolopanis** [comment]
Thanks! Looks like this indeed has had a polarization collapse at some point.

**Comment by nkern** [comment]
Oh I think I see what's going on: in the case where flags have been collapsed along polarization, then `UVFlag` re-defines polarization as a string (b/c there isn't a single polarization to assign to it as its been averaged across pols). So the issue here is that in this case, the expected type of `_polarization_array` should be `str`, not `int`.

**Comment by nkern** [comment]
So seems like `_polarization_array.expected_type` needs to be something like `(int, str)` in which case the check should check the `polarization_array.dtype` against each one a pass if any of the types agree.

**Comment by nkern** [comment]
Also, after manually changing `_polarization_array.expected_type = str` I'm now getting an error with `accepted_vals`, so seems like all of the parameter checks will need to be reconsidered.
